public/Students/Leo Jayachandran/DynamicIndex/dynamicIndexMaker.js

```diff
@@ -165,7 +165,27 @@
 					break;
 				}
 			}
+			if (canPush) {
+				let startsWithLetter = false;
+				//Create list of letters
+				let alphabet = 'ABCDEFGHIJKLMNOPQRSTUVWXYZ'.split('');
+				//Check if the name starts with any of the letters
+				for (var k = 0; k < alphabet.length; k++) {
+					// if doesn't start with a previous letter
+					if (!(startsWithLetter)) {
```

Comment:
This structure within the for loop seems a bit convoluted. I think if you start out with the startsWithLetter assignment it will make your for loop simpler and thus less error prone.

For example, I would guess that this structure would have problems with Z due to how the loop is indexing.